### PR TITLE
feat: add rawLogits option to HfRerankerConfig for cross-encoder rerankers

### PR DESCRIPTION
**Pull request by GaussAA**
## Summary

Cross-encoder reranker models (e.g. **bge-reranker-base**, bge-reranker-v2-m3) produce logits in the range of 10–20 units. When passed through `pipeline("text-classification")` the built-in sigmoid compresses all scores to ~1.0, making ranking completely flat — all candidates get nearly identical scores regardless of relevance.

This PR adds a **`rawLogits`** boolean option to `HfRerankerConfig`. When enabled, `loadRerankerPipeline` uses `AutoTokenizer` + `AutoModelForSequenceClassification` instead of `pipeline()`, and returns the raw model logits directly as scores.

### `src/search/reranker-config.ts`
- Added `rawLogits?: boolean` to `HfRerankerConfig`
- Added `PI_KNOWLEDGE_RERANKER_RAW_LOGITS` env var support (set to `"true"` to enable)
- Updated `isHfRerankerConfig` type guard to accept the new field
- Updated `rerankerCacheKey` to include the flag (both paths can coexist in cache)

### `src/model-worker.ts`
- `loadRerankerPipeline`: when `config.rawLogits` is true, imports `AutoTokenizer` + `AutoModelForSequenceClassification`, runs the model directly, and returns `[{ score: logits.data[0] }]` — same shape as the pipeline return.
- When `rawLogits` is false/undefined (default), behavior is **unchanged**.

## Usage

```bash
# Enable via environment variable
PI_KNOWLEDGE_RERANKER_RAW_LOGITS=true pi

# Or set config programmatically
{ model: "Xenova/bge-reranker-base", rawLogits: true }
```

## Verification

Tested locally with 3 Chinese medical queries against bge-reranker-base (each with 1 relevant + 2 unrelated passages). Raw logits provide **~16–18 units of separation** between relevant and irrelevant passages, vs sigmoid compressing everything to ~1.0.

```
Query: 原发性肝癌的高危人群有哪些？
  [相关] logit=  7.6909  (correctly ranked #1)
  [无关] logit=-10.1936
  [无关] logit=-10.1968
```

### CODE REVIEW

**Comment by nczz** [comment]
Thanks @GaussAA — this PR addressed a real issue with cross-encoder rerankers whose sigmoid-normalized scores flatten near 1.0.

I accepted the direction and merged it after adding a maintainer hardening pass for release readiness:

- kept the default reranker config shape unchanged when raw logits are unset
- documented `PI_KNOWLEDGE_RERANKER_RAW_LOGITS`
- added unit coverage for env parsing, cache-key separation, worker request propagation, and raw-logit scoring
- rejected multi-logit classifiers and non-finite raw scores instead of guessing a relevance label
- bumped the package to `0.7.0` and created the GitHub release

Security/privacy review found no new content egress: the local/HF raw-logit path still runs inside the isolated model worker, and API reranking behavior is unchanged.

Release: https://github.com/nczz/pi-knowledge/releases/tag/v0.7.0

Published to npm as `pi-knowledge@0.7.0`. Thanks again for the contribution.
